rubicon/objc/runtime.py

```diff
@@ -664,11 +664,39 @@ class objc_super(Structure):
 
 
 # http://stackoverflow.com/questions/3095360/what-exactly-is-super-in-objective-c
-def send_super(receiver, selName, *args, **kwargs):
+def send_super(cls, receiver, selName, *args, **kwargs):
     """Send a message named selName to the super of the receiver.
 
     This is the equivalent of [super selname:args].
+
+    Since proper 'super' semantics requires lexical scoping, the cls argument is now required.
+    It basically instructs this runtime what the lexical scope was of the caller, so that it
+    may find the appropriate superclass method implementation (if any) to call. (See issue #107)
+
+    As such, cls can be a string, which is the class name of the lexical scope of the caller,
+    such as given by the __class__.__name__ built-in in Python, or a Python 'class' type as
+    obtained from the __class__ built-in.
+
+    Eg: send_super(__class__, self, 'init') would be the typical usage pattern.
+
+    cls may also be an ObjCClass instance or a naked Class pointer.
     """
+    if not isinstance(cls, (str, ObjCClass, Class)) and not inspect.isclass(cls):
+        # Kindly remind the caller that the API has changed
+        raise TypeError("Missing/Invalid cls argument: {tp.__module__}.{tp.__qualname__} -- "
+                        + "send_super now requires its first argument be a"
+                        + " class name (str) or an instance of ObjCClass and/or Class."
+                        + " To fix this error, use Python's __class__ keyword as the first argument to"
+                        + " send_super."
+                        .format(tp=type(cls)))
+
+    if not isinstance(cls, ObjCClass):
+        if inspect.isclass(cls):
+            # __class__ was passed-in, convert to str for obj-c lookup
+            cls = cls.__name__
```

Comment:
This check isn't necessary; `__class__` is an `ObjCClass` object if you use it inside an `ObjCClass`:

```python
>>> from rubicon import objc
>>> class Foo(objc.NSObject):
...     @objc.objc_method
...     def foo(self) -> None:
...         print(type(__class__), __class__)
...         
>>> Foo.alloc().init().foo()
<class 'rubicon.objc.runtime.ObjCClass'> ObjCClass('Foo')
```

Reply:
Whaa.. ok.  Yeah that makes sense.  You Python-Fu is strong. Much respect.